entrypoint.sh

```diff
@@ -8,26 +8,34 @@ APP_GROUP=steam
 APP_HOME=/home/$APP_USER
 
 source /includes/colors.sh
+source /includes/permissions.sh
 
 if [[ "${PUID}" -eq 0 ]] || [[ "${PGID}" -eq 0 ]]; then
-    ee ">>> Running as root is not supported, please fix your PUID and PGID!"
+    ee ">>> Running Palworld as root is not supported, please fix your PUID and PGID!"
     exit 1
 elif [[ "$(id -u steam)" -ne "${PUID}" ]] || [[ "$(id -g steam)" -ne "${PGID}" ]]; then
     ew "> Current $APP_USER user PUID is '$(id -u steam)' and PGID is '$(id -g steam)'"
     ew "> Setting new $APP_USER user PUID to '${PUID}' and PGID to '${PGID}'"
     groupmod -g "${PGID}" "$APP_GROUP" && usermod -u "${PUID}" -g "${PGID}" "$APP_USER"
-else 
+else
     ew "> Current $APP_USER user PUID is '$(id -u steam)' and PGID is '$(id -g steam)'"
     ew "> PUID and PGID matching what is requested for user $APP_USER"
 fi
 
-chown -R "$APP_USER":"$APP_GROUP" "$APP_HOME"
-chown -R "$APP_USER":"$APP_GROUP" "$GAME_ROOT"
-chown "$APP_USER":"$APP_GROUP" /entrypoint.sh
-chown "$APP_USER":"$APP_GROUP" /PalWorldSettings.ini.template
-chown -R "$APP_USER":"$APP_GROUP" /scripts
-chown -R "$APP_USER":"$APP_GROUP" /includes
+change_ownership $APP_USER $APP_GROUP "$APP_HOME"
+change_ownership $APP_USER $APP_GROUP "$GAME_ROOT"
+change_ownership $APP_USER $APP_GROUP /entrypoint.sh
+change_ownership $APP_USER $APP_GROUP /PalWorldSettings.ini.template
+change_ownership $APP_USER $APP_GROUP /scripts
+change_ownership $APP_USER $APP_GROUP /includes
 
-ew_nn "> id steam: " ; e "$(id steam)"
+ew_nn "> id steam: "
+e "$(id steam)"
 
-exec gosu $APP_USER:$APP_GROUP "$@"
+if [ "$(id -u)" -eq 0 ]; then
```

Comment:
This is already covered in the if above this, why add it again?
The user-jail works as good because we use exec in combination with gosu, there should be no else part needed.

Dont like this part, please explain the idea of this part.

Reply:
This was to allow people to run it as not root since gosu did not like it when it tried to run as the $APP_USER:$APP_GROUP that it already was. Perhaps I should have separated the PRs. One that improves error messages and one that allows for non-root running. I personally run it as root but I figured might as well add that feature since I (at least thought I) saw how.

Reply:
Haha yea, I guess I let this one get away from me a bit. This should have been 3 or 4
1. `thread` -> `threat`
2. Improved error message for when trying to change permissions but not running as root
3. Improved error message for PUID/PGID
4. Ability to run as non-root